### PR TITLE
Added secrets section in Kubernetes Task

### PR DESCRIPTION
**Pull request by Anumita**
Docker registry secret
secreets

Generic secret
image

### CODE REVIEW

File: Tasks/Kubernetes/task.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "version": {
         "Major": 0,
         "Minor": 1,
-        "Patch": 14
+        "Patch": 16
```
Comment (Anumita) [inline Tasks/Kubernetes/task.json]: will change this to 15. Please ignore

File: Tasks/Kubernetes/Strings/resources.resjson/en-US/resources.resjson [hunk 2]
```diff
@@ -4,11 +4,13 @@
   "loc.description": "Deploy, configure, update your Kubernetes cluster in Azure Container Service by running kubectl commands.",
   "loc.instanceNameFormat": "kubectl $(command)",
   "loc.group.displayName.commands": "Commands",
-  "loc.group.displayName.registry": "Container Registry Details",
+  "loc.group.displayName.secrets": "Secrets",
   "loc.group.displayName.advanced": "Advanced",
   "loc.group.displayName.output": "Output",
   "loc.input.label.kubernetesServiceEndpoint": "Kubernetes Service Connection",
   "loc.input.help.kubernetesServiceEndpoint": "Select a Kubernetes service connection.",
+  "loc.input.label.namespace": "Namespace",
```
Comment (bansalaseem) [inline Tasks/Kubernetes/Strings/resources.resjson/en-US/resources.resjson]: Why this change?

File: Tasks/Kubernetes/src/kubernetessecret.ts [hunk 3]
```diff
@@ -8,19 +8,32 @@ import ClusterConnection from "./clusterconnection";
...
+    else if (typeOfSecret === "generic")
+    {
+        return createGenericSecret(connection, secret);
     }
 }
 
 function deleteSecret(connection: ClusterConnection, secret: string): any {
     tl.debug(tl.loc('DeleteSecret', secret));
     var command = connection.createCommand();
+    command.arg(getNameSpace());
```
Comment (bansalaseem) [inline Tasks/Kubernetes/src/kubernetessecret.ts]: why this change?

File: Tasks/Kubernetes/src/kubernetessecret.ts [hunk 4]
```diff
@@ -33,12 +46,13 @@ function deleteSecret(connection: ClusterConnection, secret: string): any {
...
 }
 
-function createSecret(connection: ClusterConnection, authenticationToken: AuthenticationToken, secret: string): any {
+function createDockerRegistrySecret(connection: ClusterConnection, authenticationToken: AuthenticationToken, secret: string): any {
 
     if(authenticationToken)
     {
         tl.debug(tl.loc('CreatingSecret', secret));
         var command = connection.createCommand();
+        command.arg(getNameSpace());
```
Comment (bansalaseem) [inline Tasks/Kubernetes/src/kubernetessecret.ts]: why this change?

File: Tasks/Kubernetes/task.json [hunk 5]
```diff
@@ -47,6 +47,14 @@
             "label": "Kubernetes Service Connection",
             "helpMarkDown": "Select a Kubernetes service connection."
         },
+        {
+            "name": "namespace",
```
Comment (bansalaseem) [inline Tasks/Kubernetes/task.json]: Why this change?

File: Tasks/Kubernetes/task.json [hunk 6]
```diff
@@ -200,15 +236,6 @@
             "groupName": "advanced",
             "visibleRule": "versionOrLocation = location"
         },
-        {
-            "name": "namespace",
-            "type": "string",
-            "label": "Namespace",
-            "required": false,
```
Comment (bansalaseem) [inline Tasks/Kubernetes/task.json]: why this change? Please undo it

File: Tasks/Kubernetes/Strings/resources.resjson/en-US/resources.resjson (hunk 2 above)
Comment (Anumita) [reply Tasks/Kubernetes/Strings/resources.resjson/en-US/resources.resjson]: As discussed with Atul ,the commands and secrets section use the same namespace field so it would be more readable if the namespace field is moved just below the kubernetes service connection field.

File: Tasks/Kubernetes/task.json [hunk 7]
```diff
@@ -112,52 +147,53 @@
...
             "name": "azureSubscriptionEndpoint",
             "aliases": ["azureSubscription"],
             "type": "connectedService:AzureRM",
             "label": "Azure subscription",
             "helpMarkDown": "Select an Azure subscription, which has your Azure Container Registry.",
-            "visibleRule": "containerRegistryType = Azure Container Registry",
+            "visibleRule": "secretType = dockerRegistry && containerRegistryType = Azure Container Registry",
             "defaultValue": "",
-            "groupName": "registry"
+            "groupName": "secrets"
```
Comment (jikuma) [inline Tasks/Kubernetes/task.json]: Changing group of an old input will not break the old task right?. Have we verified it.

Comment (Anumita) [reply Tasks/Kubernetes/task.json]: Yes, i have verified. The task will run the same
